controllers/metallb_controller_test.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/metallb/metallb-operator/api/v1alpha1"
+	"github.com/metallb/metallb-operator/test/consts"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("MetalLB Controller", func() {
+	Context("syncMetalLB", func() {
+		metallb := &v1alpha1.Metallb{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "metallb",
+				Namespace: consts.MetallbNameSpace,
+			},
+		}
+		AfterEach(func() {
+			err := k8sClient.Delete(context.Background(), metallb)
+			if err != nil {
+				if !apierrors.IsNotFound(err) {
+					Fail(err.Error())
+				}
```

Comment:
We should also clean up all the items created by test test: DaemonSet, Deployment, ...
OwnerRef will not work in envtest since no default controllers are running

Reply:
done